src/Network/Consul.hs

```diff
@@ -195,3 +504,110 @@ withSequencer client sequencer action lostAction delay dc =
       case valid of
         True -> pulseLock
         False -> lostAction
+
+
+{- Agent -}
+{-getHealthChecks :: MonadIO m => Manager -> Text -> PortNumber -> Maybe Datacenter -> m [Check]
+getHealthChecks  manager hostname portNumber dc = do
+  request <- createRequest hostname portNumber "/agent/checks" Nothing Nothing False dc
+ -}
+
+registerHealthCheck :: MonadIO m => ConsulClient -> RegisterHealthCheck -> m ()
+registerHealthCheck client@ConsulClient{..} request = do
+  let hostnameWithScheme = hostWithScheme client
+  initReq <- liftIO $ parseUrlThrow $ T.unpack $ T.concat [hostnameWithScheme, ":", T.pack $ show ccPort ,"/v1/agent/check/register"]
+  let httpReq = initReq { method = "PUT", requestBody = RequestBodyBS $ BL.toStrict $ encode request}
+  liftIO $ withResponse httpReq ccManager $ \ response -> do
+    _bodyParts <- brConsume $ responseBody response
+    return ()
+
+deregisterHealthCheck :: MonadIO m => ConsulClient -> Maybe Datacenter -> Text -> m ()
+deregisterHealthCheck client@ConsulClient{..} dc checkId = do
+  let hostnameWithScheme = hostWithScheme client
+  initReq <- createRequest hostnameWithScheme
+                           ccPort
+                           (T.concat ["/v1/agent/check/deregister/", checkId])
+                           Nothing
+                           Nothing
+                           False
+                           dc
+  liftIO $ withResponse initReq ccManager $ \ response -> do
+    _bodyParts <- brConsume $ responseBody response
+    return ()
+
+
+passHealthCheck :: MonadIO m => ConsulClient -> Text -> Maybe Datacenter -> m ()
+passHealthCheck client checkId dc = do
+  -- Using `Just ""` as the `body` to ensure a PUT request is used.
+  -- Consul < 1.0 accepted a GET here (which was a legacy mistake).
+  -- In 1.0, they switched it to require a PUT.
+  -- See also:
+  --   * https://github.com/hashicorp/consul/issues/3659
+  --   * https://github.com/cablehead/python-consul/pull/182
+  --   * https://github.com/hashicorp/consul/blob/51ea240df8476e02215d53fbfad5838bf0d44d21/CHANGELOG.md
+  --     Section "HTTP Verbs are Enforced in Many HTTP APIs":
+  --     > Many endpoints in the HTTP API that previously took any HTTP verb
+  --     > now check for specific HTTP verbs and enforce them.
+  let portNumber = ccPort client
+      manager = ccManager client
+      -- !QUESTION: do we use the checkId for the hostname?
+      hostname = hostWithScheme client
+      --hostname = checkId
+  initReq <- createRequest hostname
+                           portNumber
+                           (T.concat ["/v1/agent/check/pass/", checkId])
+                           Nothing
+                           (Just "")
+                           False
+                           dc
+  -- TODO: see warning: `Defined but not used: ‘response’`
```

Comment:
Can be underscored (`_response`) for now; long term we should naturally inspect it for success.

Reply:
Great point!